core/common/src/main/java/alluxio/resource/DynamicResourcePool.java

```diff
@@ -181,26 +178,15 @@ public static Options defaultOptions() {
   private final int mMaxCapacity;
   private final int mMinCapacity;
 
-  // Tracks the resources that are available ordered by lastAccessTime (the first one is
+  // Tracks the resources that are available ordered by lastAccessTime (the head is
   // the most recently used resource).
   @GuardedBy("mLock")
-  private TreeSet<ResourceInternal<T>> mResourceAvailable =
-      new TreeSet<>(new Comparator<ResourceInternal<T>>() {
-        @Override
-        public int compare(ResourceInternal<T> c1, ResourceInternal<T> c2) {
-          if (c1 == c2) {
-            return 0;
-          }
-          if (c1.mLastAccessTimeMs == c2.mLastAccessTimeMs) {
-            return c1.mIdentity - c2.mIdentity;
-          }
-          return (int) (c2.mLastAccessTimeMs - c1.mLastAccessTimeMs);
-        }
-      });
+  private Deque<ResourceInternal<T>> mResourceAvailable;
 
   // Tracks all the resources that are not closed.
```

Comment:
Is this accurate? Should this be `Tracks all resources managed by this resource pool`? I'm just wondering if you can `close()` a resource in the pool and if this data structure actually updates to reflect that.